build-ps/jenkins/build-binary

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o xtrace
+
+
+# ------------------------------------------------------------------------------
+# Declate all input variables
+# ------------------------------------------------------------------------------
+JOB_CMAKE=${JOB_CMAKE:-cmake}
+COMPILER=${COMPILER:-default}
+CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE:-RelWithDebInfo}
+WITH_TOKUDB=${WITH_TOKUDB:-ON}
+WITH_ROCKSDB=${WITH_ROCKSDB:-ON}
+WITH_EMBEDDED_SERVER=${WITH_EMBEDDED_SERVER:-ON}
+WITH_RAPID=${WITH_RAPID:-ON}
+WITH_SSL=${WITH_SSL:-system}
+WITHOUT_PERFSCHEMA_STORAGE_ENGINE=${WITHOUT_PERFSCHEMA_STORAGE_ENGINE:-OFF}
+ANALYZER_OPT=${ANALYZER_OPT:-}
+MAKE_OPTS=${MAKE_OPTS:--j$(grep -c ^processor /proc/cpuinfo) VERBOSE=1}
+CMAKE_OPTS=${CMAKE_OPTS:-}
+BUILD_COMMENT=${BUILD_COMMENT:-}
+TAG=${TAG:-}
+DIST_NAME=${DIST_NAME:-}
+SSL_VER=${SSL_VER:-}
+
+
+# ------------------------------------------------------------------------------
+# set working dir
+# ------------------------------------------------------------------------------
+SOURCEDIR=$(cd $(dirname $0)/../..; pwd -P)
+WORKDIR=$(cd ${1:-.}; pwd -P)
+INSTALL_DIR=${SOURCEDIR}/DESTDIR
+mkdir -p ${WORKDIR} ${INSTALL_DIR} ${WORKDIR}/libboost 
+
+
+# ------------------------------------------------------------------------------
+# Set OS/Arch flags
+# ------------------------------------------------------------------------------
+if [ -n "$(which rpm)" ]; then
+    TARGET_CFLAGS+=" $(rpm --eval %optflags | sed -e 's|march=i386|march=i686|g')"
+fi
+if [[ $(getconf LONG_BIT) == 32 ]]; then
+    TARGET_ARCH="i686"
+    TARGET_CFLAGS+=" -m32 -march=i686"
+else
+    TARGET_ARCH="$(uname -m)"
+fi
+
+
+# ------------------------------------------------------------------------------
+# Set Debug options
+# ------------------------------------------------------------------------------
+if [[ $CMAKE_BUILD_TYPE == Debug ]]; then
+    BUILD_COMMENT+="-debug"
+    TARGET_CFLAGS=$(
+        echo " ${TARGET_CFLAGS} " | \
+            sed -e 's/ -O[0-9]* / /' \
+                -e 's/-Wp,-D_FORTIFY_SOURCE=2/ /' \
+                -e 's/ -unroll2 / /' \
+                -e 's/ -ip / /'
+    )
+    CMAKE_OPT+=" -DMYSQL_MAINTAINER_MODE=ON -DDEBUG_EXTNAME=OFF"
```

Comment:
`MYSQL_MAINTAINER_MODE` must be set to `ON` unconditionally (regardless of Debug/Release configuration)

Reply:
Not sure about `-DDEBUG_EXTNAME=OFF`:
if it's not specified the primary Percona Server executable will be named `mysqld-debug`
if it is, then just `mysqld`.
I cannot tell what user intention is here.

Reply:
`MYSQL_MAINTAINER_MODE ` - fixed
`DEBUG_EXTNAME` - @EvgeniyPatlan asked to leave as is